pta-intelligent-search-api/src/main/java/fi/maanmittauslaitos/pta/search/api/search/OntologyElasticsearchQueryProviderImpl.java

```diff
@@ -248,6 +253,28 @@ private QueryBuilder freetextQuery(String field, String word, double weight, dou
 		return query;
 	}
 
+    private void addSynonymQueries(Collection<String> searchTerms, BoolQueryBuilder boolQuery) {
+        for (String searchTerm : createSynonymList(searchTerms)) {
+            addFreeTextQuery(searchTerm, boolQuery);
+        }
+    }
+
+    private List<String> createSynonymList(Collection<String> searchTerms) {
+        List<String> synonymList = new ArrayList<>();
+        for(String term : searchTerms) {
+            List<String> termSynonyms = synonymsWords.get(term);
+            if(termSynonyms != null && termSynonyms.size() > 0) {
+                for (String synonym : termSynonyms) {
+                	if(synonym != term) {
+						synonymList.add(synonym);
+					}
+                }
+                //synonymList.remove(term);
```

Comment:
Why is this commented? My intuition is that the original search term that was expanded to synonyms should not appear as a synonym to itself.